### DISCUSSION TITLE
Never should be selected as restartpolicy of prepare job.

### DISCUSSION THREAD

**Discussion by fengjiankui121** [rook/rook#10228] [Ideas]
After the prepare job fails, the pod will be deleted because the currently selected restartpolicy is onFailure, which causes the failure to be located. So, I think Nerver should be selectd as restartpolicy.

**Comment by travisn** [comment]
Why not restart the job on failure? What do you mean "the failure to be located"?

**Reply by fengjiankui121** [reply]
@travisn  Currently, the restartpolicy  of prepare job is onFailure. This restartpolicy  will delete the pod after the job fails, which will cause the log of the job to be lost. After the restartpolicy  is set to Nerver, if the job fails, k8s will create a new pod, but will not delete the failed pod and keep the log information. Therefore, it is recommended to set the restart policy of the prepare job to Nerver.

**Reply by travisn** [reply]
Could you open an issue for this request? If you would like to submit a PR for this, please also see the [Contributor guide](https://rook.github.io/docs/rook/v1.9/development-flow.html). Thanks!
